### PR TITLE
opensearch 1.0.0-beta1 (new formula)

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Modified from the `elasticsearch` formula. Builds and passes tests. I realise this is still a beta release but I think we should include it sooner rather than later to give people time to migrate from `elasticsearch`.

### CODE REVIEW

File: Formula/opensearch.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  url "https://github.com/opensearch-project/OpenSearch/archive/refs/tags/1.0.0-beta1.tar.gz"
+  sha256 "d23385aa42f636049ae270bdb496843dc8d2dfd88bd7f4761e305e8193b76399"
+  license "Apache-2.0"
+
+  depends_on "gradle@6" => :build
+  depends_on "openjdk"
+
+  def cluster_name
+    "opensearch_#{ENV["USER"]}"
+  end
```
Comment (SMillerDev) [inline Formula/opensearch.rb:13]: wouldn't this always be `opensearch_brew` because CI?

File: Formula/opensearch.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          <key>WorkingDirectory</key>
+          <string>#{var}</string>
+          <key>StandardErrorPath</key>
+          <string>#{var}/log/opensearch.log</string>
+          <key>StandardOutPath</key>
+          <string>#{var}/log/opensearch.log</string>
+        </dict>
+      </plist>
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/opensearch.rb:106]: Can this use a `service do` block

**Review by SMillerDev — COMMENTED** [review]
Since this is backed by Amazon and the beta is based on existing software I'm fine shipping this.

File: Formula/opensearch.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Formula/opensearch.rb:13]: Technically, guess so. Not when built from source, though. That naming seems fine anyway though, right?

Comment (SMillerDev) [reply Formula/opensearch.rb]: yeah, but in that case it hardly needs to be a method IYAM.

**Pull request closed without merging**